qcodes/dataset/data_set_protocol.py

```diff
@@ -465,9 +484,25 @@ def _export_file_name(self, prefix: str, export_type: DataExportType) -> str:
 
     def _export_as_netcdf(self, path: Path, file_name: str) -> Path:
         """Export data as netcdf to a given path with file prefix"""
+        import xarray as xr
         file_path = path / file_name
-        xarr_dataset = self.to_xarray_dataset()
-        xarray_to_h5netcdf_with_complex_numbers(xarr_dataset, file_path)
+        if self._estimate_ds_size() > 1000:
```

Comment:
Should the limit be configurable. Should we be able to disable this

Reply:
yes. maybe.